pkg/machine/shim/host.go

```diff
@@ -840,3 +840,59 @@ func validateDestinationPaths(dest string) error {
 	}
 	return nil
 }
+
+// InspectRunningMachine finds and returns information about the currently running machine.
+// It searches across all VM providers to find a machine in the running state.
+// Returns nil if no running machine is found.
+func InspectRunningMachine(vmstubbers []vmconfigs.VMProvider, _ machine.ListOptions) (*machine.InternalInspectInfo, error) {
```

Comment:
I don't think this is right or future proof, wsl machines support multiple running machines at the same time and we might want to have that for other providers as well.

I think the checking the ssh connection info like done in cmd/podman/compose_machine.go is likely the better thing to match the machine vs connection. the bindings client.URI should show the URL